mbr.go

```diff
@@ -77,7 +77,7 @@ func (this *MBR) Check() error {
 		}
 
 		// Check if partition last sector out of uint32 bounds
-		if uint64(lp.GetLBAStart())+uint64(lp.GetLBALen()) > uint64(0xFFFFFFFF) {
+		if uint64(lp.GetLBAStart())+uint64(lp.GetLBALen()) > uint64(0xFFFFFFFF)+1 {
```

Comment:
https://en.wikipedia.org/wiki/Master_boot_record
The organization of the partition table in the MBR limits the maximum addressable storage space of a partitioned disk to 2 TiB (232 × 512 bytes).

I thing about default behavior must max follow standard.
And I thing about low level library may have tools for out of standard when it need, but it must be manual and when library user know what he do.

I mean about MBR extension (such as +1 byte for gdisk compatible, or disable max byte check (for allow create 2TB partition with 2TB offset) must be enabled by flag.

I suggest to add additional fields to MBR struct for customize checks:
1. AllowableOverflow - for can add a byte for gdisk compatible (and check - if GDisk add exactly one byte always - may add method GDiskCompatible for set value of the field to one).
2. Add method Read to *MBR for read itself with additinal settings.
3. Change current MBR read method - for create empty mbr object, then call mbr.Read(disk) and return &mbr, err